### PR TITLE
chore(j-s): Show send to prison admin file

### PR DESCRIPTION
**Pull request by oddsson**
# Show send to prison admin file

[Asana](https://app.asana.com/0/1199153462262248/1208888959002469/f)

## What

When public prosecutor staff send cases to prison admin, they can upload a file with the request. That has been implemented but we forgot to show the uploaded file in file lists.

## Screenshots / Gifs

Screenshot 2024-12-05 at 14 49 48

## Checklist:

- [x] I have performed a self-review of my own code
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] Formatting passes locally with my changes
- [x] I have rebased against main before asking for a review

## Summary by CodeRabbit

- **New Features**
	- Added a new message for the "sent to prison admin" section in the Indictment Case Files List.
	- Introduced a modular `FileSection` component for improved case file organization.
	- Implemented hooks for filtering case files and managing user permissions.

- **Bug Fixes**
	- Enhanced user permission checks for viewing specific case file categories.

- **Refactor**
	- Improved the structure and clarity of the Indictment Case Files List component.

### CODE REVIEW

File: apps/judicial-system/web/src/components/IndictmentCaseFilesList/IndictmentCaseFilesList.tsx [hunk 1]
```diff
@@ -267,14 +303,24 @@ const IndictmentCaseFilesList: FC<Props> = ({
...
+              onOpenFile={onOpen}
+            />
+          </Box>
+        )}
+      <FileSection
+        title={formatMessage(strings.sentToPrionsAdmin)}
+        files={filteredFiles.sentToPrisonAdminFiles}
+        onOpenFile={onOpen}
+        shouldRender={permissions.canViewSentToPrisonAdminFiles}
+      />
```
[the remark this answers is not in the document]
Comment (oddsson) [reply apps/judicial-system/web/src/components/IndictmentCaseFilesList/IndictmentCaseFilesList.tsx]: Thanks, fixed

File: apps/judicial-system/web/src/components/IndictmentCaseFilesList/IndictmentCaseFilesList.tsx [hunk 2]
```diff
@@ -201,34 +237,34 @@ const IndictmentCaseFilesList: FC<Props> = ({
...
-          {courtRecords && courtRecords.length > 0 && (
-            <RenderFiles caseFiles={courtRecords} onOpenFile={onOpen} />
-          )}
+          {filteredFiles.courtRecords &&
+            filteredFiles.courtRecords.length > 0 && (
+              <RenderFiles
+                caseFiles={filteredFiles.courtRecords}
+                onOpenFile={onOpen}
+              />
+            )}
```
Comment (unakb) [inline apps/judicial-system/web/src/components/IndictmentCaseFilesList/IndictmentCaseFilesList.tsx:251]: Nice! Did you consider also moving the rulings check below to the filePersmissions handling?

**Review by unakb — APPROVED** [review]
Nice change! One thought

File: apps/judicial-system/web/src/components/IndictmentCaseFilesList/IndictmentCaseFilesList.tsx (hunk 2 above)
Comment (oddsson) [reply apps/judicial-system/web/src/components/IndictmentCaseFilesList/IndictmentCaseFilesList.tsx:259]: Thanks! I've added a new permission for ruling files 🙌

**Pull request merged**